### PR TITLE
New Packages: gearshifft, clFFT

### PR DESCRIPTION
**Pull request by ax3l**
Adds a [gearshifft](https://github.com/mpicbg-scicomp/gearshifft) package, a benchmark suite for heterogeneous implementations of FFTs.

Adds [clFFT](https://github.com/clMathLibraries/clFFT), a software library containing FFT functions written in OpenCL.

### CODE REVIEW

**Comment by ax3l** [comment]
ccing @tdd11235813 @psteinb first draft of your package for spack.
if this works, you can install as easy as `spack install gearshifft`

Full install instructions for first-timers (have the latest cuda & opencl drivers installed beforehand!):
```bash
# have modules installed beforehand, e.g. via:
sudo apt-get install lmod

# clone spack
git clone https://github.com/ax3l/spack.git --branch=topic-gearshifft
cd spack/bin

# build gearshifft
./spack install gearshifft

# load it
module use ../share/spack/lmod/<putYourOSHere>/Core
module av
# module load ...
```

**Comment by adamjstewart** [comment]
Let me know when this PR is no longer a WIP.

**Comment by ax3l** [comment]
Thanks will do, upstream is going to push a fixed release in the next days. I'll let you know when I am finished.

**Comment by ax3l** [comment]
the package works so far for all but OpenCL which is not found (clFFT is build but gearshifft does not find it). This might be related to #4669 or an issue in gearshifft's FindOpenCL package

**Comment by ax3l** [comment]
The package works for now.
It's currently missing the OpenCL benchmarks due to one package issue in pocl (at least for me) and one in the upstream CMake scripts. Both do not block the overall package, but it's a bit nasty that it exposes OpenCL without providing it. Let me at least comment it out for now.

File: var/spack/repos/builtin/packages/gearshifft/package.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    #variant('clfft', default=True,
+    #        description='Compile gearshifft_clfft')
+    variant('fftw', default=True,
+            description='Compile gearshifft_fftw')
+    variant('openmp', default=True,
+            description='use OpenMP parallel fftw libraries')
+    # variant('hcfft', default=True,
+    #         description='Not implemented yet')
+
+    # depends_on C++14 compiler, e.g. GCC 5.0+
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/gearshifft/package.py]: @alalazo btw, is there a way to express C++14 requirements already? :)

Comment (alalazo) [inline var/spack/repos/builtin/packages/gearshifft/package.py]: Currently not...

**Comment by ax3l** [comment]
@adamjstewart finished for a first package :)

File: var/spack/repos/builtin/packages/gearshifft/package.py (hunk 1 above)
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gearshifft/package.py]: You could say:
```python
conflicts('%gcc@:4')
```
but you'd have to do it for every compiler. Maybe we can find a way to express this requirement when we rework compiler support.

Comment (ax3l) [inline var/spack/repos/builtin/packages/gearshifft/package.py]: @adamjstewart ok, so I'll leave it that way for now.

**Comment by tdd11235813** [comment]
Thank you so much everyone, especially to @ax3l for getting gearshifft into spack! :)

**Pull request merged**